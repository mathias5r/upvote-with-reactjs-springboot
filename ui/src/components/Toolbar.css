
.button-toolbar{
	margin-top: 10px;
	width: 100%;
	align-content: center;
	display: flex;
	justify-content: center
}


.button{
	width: 10%;
	margin: 10px;
}