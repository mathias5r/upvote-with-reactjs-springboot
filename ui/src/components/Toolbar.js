import React from 'react';
import ReactDOM from 'react-dom';
import { Component } from 'react';
import PostView from './PostView'
import AddView from './AddView'
import {Button, ButtonGroup} from 'react-bootstrap'
import './Toolbar.css';

class Toolbar extends Component{

	constructor(props, context) {
	    super(props, context);

	    this.handleHomeClick = this.handleHomeClick.bind(this);
	    this.handleAddClick = this.handleAddClick.bind(this);
  	}

	handleHomeClick() {
		ReactDOM.render(<PostView />, document.getElementById('body'));
	}

	handleAddClick() {
		ReactDOM.render(<AddView />, document.getElementById('body'));
	}


	render() {
    	return (
    		<ButtonGroup>
				<Button  onClick={this.handleHomeClick} >Home</Button>
				<Button  onClick={this.handleAddClick}>Adicionar</Button>
			</ButtonGroup>
    	);
  	}
}
export default Toolbar;