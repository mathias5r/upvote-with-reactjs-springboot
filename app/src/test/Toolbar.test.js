import React from 'react';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Toolbar from '../components/Toolbar'
import ShallowRenderer from 'react-test-renderer/shallow';;
Enzyme.configure({ adapter: new Adapter() });

describe('Testing PostView.js', () =>{
	it('should render correctly...', () =>{
		const renderer = new ShallowRenderer();
		renderer.render(<Toolbar />);
	  	const tree = renderer.getRenderOutput();
  		expect(tree).toMatchSnapshot();
	});
});