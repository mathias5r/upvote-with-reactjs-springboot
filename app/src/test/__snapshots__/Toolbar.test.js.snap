// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostView.js should render correctly... 1`] = `
<ButtonGroup
  block={false}
  bsClass="btn-group"
  justified={false}
  vertical={false}
>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="home"
    disabled={false}
    onClick={[Function]}
  >
    Home
  </Button>
  <Button
    active={false}
    block={false}
    bsClass="btn"
    bsStyle="default"
    className="adicionar"
    disabled={false}
    onClick={[Function]}
  >
    Adicionar
  </Button>
</ButtonGroup>
`;
